tools/common/Driver.cs

```diff
@@ -126,6 +126,20 @@ static void AddSharedOptions (Application app, Mono.Options.OptionSet options)
 			options.Add ("root-assembly:", "Specifies any root assemblies. There must be at least one root assembly, usually the main executable.", (v) => {
 				app.RootAssemblies.Add (v);
 			});
+			options.Add ("optimize=", "A comma-delimited list of optimizations to enable/disable. To enable an optimization, use --optimize=[+]remove-uithread-checks. To disable an optimizations: --optimize=-remove-uithread-checks. Use '+all' to enable or '-all' disable all optimizations. Only compiler-generated code or code otherwise marked as safe to optimize will be optimized.\n" +
+					"Available optimizations:\n" +
+					"    dead-code-elimination: By default always enabled (requires the linker). Removes IL instructions the linker can determine will never be executed. This is most useful in combination with the inline-* optimizations, since inlined conditions almost always also results in blocks of code that will never be executed.\n" +
+					"    remove-uithread-checks: By default enabled for release builds (requires the linker). Remove all UI Thread checks (makes the app smaller, and slightly faster at runtime).\n" +
+					"    inline-isdirectbinding: By default enabled (requires the linker). Tries to inline calls to NSObject.IsDirectBinding to load a constant value. Makes the app smaller, and slightly faster at runtime.\n" +
+					"    inline-intptr-size: By default enabled for builds that target a single architecture (requires the linker). Inlines calls to IntPtr.Size to load a constant value. Makes the app smaller, and slightly faster at runtime.\n" +
+#if MONOTOUCH
+					"    inline-runtime-arch: By default always enabled (requires the linker). Inlines calls to ObjCRuntime.Runtime.Arch to load a constant value. Makes the app smaller, and slightly faster at runtime.\n",
+#else
+					"",
```

Comment:
?

Reply:
The previous non-conditional line (https://github.com/xamarin/xamarin-macios/pull/3242/files/59c94509936e4116c1c13216fd840d6fd98e219e#diff-41355d95fe45cbbf5ce00dd86c5bd41dR134) ends with a `+`, which means that a string must follow in all cases.

I could probably reorder the lines so that the `#if MONOTOUCH` isn't the last line, in which case this would not be needed.